### DISCUSSION TITLE
How to finetune chronos-t5-small using GluonTS datasets?

### DISCUSSION THREAD

**Discussion by kushalkodn-db** [amazon-science/chronos-forecasting#152] [Q&A] [answered]
Hi, I understand how to evaluate a model using a GluonTS dataset. However, do you have any example scripts on how to fine-tune a Chronos model using one of the GluonTS datasets? I see that the chronos config files require a path to an arrow file, but I don't see how to generate this arrow file out of a GluonTS dataset. 

Second, I have a clarification question about the reported fine-tuning results. For the following results, which of the approaches did you use?
1. Fine-tune `amazon/chronos-t5-small` 27 separate times (once for each of the zero-shot train datasets), compute 27 different MASE/WQL scores (one for each of the zero-shot test datasets), and then aggregate the metrics with the geometric mean to produce those results?
2. Fine-tune `amazon/chronos-t5-small` once on a large combined dataset (all 27 zero-shot train datasets combined), and then compute a single MASE/WQL score at the end.

<img width="398" alt="image" src="https://github.com/user-attachments/assets/290b17e5-6247-4251-9dee-1ee4d44d24c5">

Thanks in advance!

**Comment by abdulfatir** [comment, marked as answer]
@kushalkodn-db Everything is described here: https://github.com/amazon-science/chronos-forecasting/tree/main/scripts
You will need to convert your data into GluonTS arrow format first as described there.

We used the first approach, i.e., dataset-specific fine-tuning.

**Reply by KushKod04** [reply]
Yes, that totally makes sense. I have been able to fine-tune chronos on my own datasets by converting csv->arrow. My question is more so about how I would convert a GluonTS Dataset into an arrow file. I'm trying to use the same 27 Benchmark 2 datasets your team used, in order to finetune Chronos with each dataset individually.

Thanks for clarifying your finetuning process!
